eva/executor/function_scan_executor.py

```diff
@@ -37,14 +39,41 @@ def __init__(self, node: FunctionScanPlan):
     def validate(self):
         pass
 
+    def persistCost(self, name, type, frame_count, resolution, time):
+        # TODO: compute cost using time
+        cost = time
+        catalog_manager = CatalogManager()
+        catalog_manager.insert_udf_cost_catalog_entry(
+            name, type, cost, frame_count, resolution
+        )
+
     def exec(self, *args, **kwargs) -> Iterator[Batch]:
         assert (
             "lateral_input" in kwargs
         ), "Key lateral_input not passed to the FunctionScan"
         lateral_input = kwargs.get("lateral_input")
+        catalog_manager = CatalogManager()
         try:
             if not lateral_input.empty():
+                t_start = time()
                 res = self.func_expr.evaluate(lateral_input)
+                t_end = time()
+                time_taken = t_end - t_start
+                resolution = None
+                udf_obj = catalog_manager.get_udf_catalog_entry_by_name(
+                    self.func_expr.name
+                )
+                if udf_obj:
+                    type = udf_obj.type
+                else:
+                    type = None
+                self.persistCost(
```

Comment:
This will add multiple entries (for each iteration) to the catalog table, right? Is it expected?

Reply:
Yes for now. Will make edits in the future